lib/ansible/modules/database/aerospike/aerospike_migrations.py

```diff
@@ -0,0 +1,535 @@
+        if len(cluster_key) == 1:
+            return True
+        return False
+
+    # checks a few things to make sure we're OK to say the cluster has no migs
+    # it could be in a unhealthy condition that does not allow migs, or
+    # a split brain
+    def _cluster_good_state(self):
+        if self._cluster_key_consistent() is not True:
+            return False, "Cluster key inconsistent."
+        if self._is_min_cluster_size() is not True:
+            return False, "Cluster min size not reached."
+        if self._cluster_migrates_allowed() is not True:
+            return False, "migrate_allowed is false somewhere."
+        return True, "OK."
+
+    # public function
+    # returns a bool, modifiable behavior by module.params
+    def has_migs(self, local=True):
+        """returns a boolean, False if no migrations otherwise True"""
+        consecutive_good = 0
+        try_num = 0
+        skip_reason = list()
+        while \
+            try_num < int(self.module.params['tries_limit']) and \
+                consecutive_good < \
+                int(self.module.params['consecutive_good_checks']):
+
+            self._update_nodes_list()
+            self._update_cluster_statistics()
+
+            # These checks are outside of the while loop because
+            # we probably want to skip & sleep instead of failing entirely
+            stable, reason = self._cluster_good_state()
+            if stable is not True:
+                skip_reason.append(
+                    "Skipping on try#" + str(try_num) +
+                    " for reason:" + reason
+                )
+            else:
+                if self._can_use_cluster_stable() is True:
+                    if self._cluster_stable() is True:
+                        consecutive_good += 1
+                    else:
+                        consecutive_good = 0
+                        skip_reason.append(
+                            "Skipping on try#" + str(try_num) +
+                            " for reason:" + " cluster_stable"
+                        )
+                elif self._has_migs(local) is True:
+                    # print("_has_migs")
+                    skip_reason.append(
+                        "Skipping on try#" + str(try_num) +
+                        " for reason:" + " migrations"
+                    )
+                    consecutive_good = 0
+                elif self._has_migs(local) is False:
+                    consecutive_good += 1
+                    if consecutive_good is self.module.params[
+                            'consecutive_good_checks']:
+                        break
+            try_num += 1
+            sleep(self.module.params['sleep_between_checks'])
+            # print(skip_reason)
+        if consecutive_good is self.module.params['consecutive_good_checks']:
+            return False, None
+        return True, skip_reason
+
+
+def main():
+    """main method for ansible module"""
+    run_module()
```

Comment:
Why don't you put the contents of `run_module()` into `main()`? Is there an advantage of having this additional indirection?

Reply:
Just was following the template, wasn't sure if there was a dependency on that method being there. Think I should move it?

Reply:
Ah, I didn't knew there was a template :) Well, in that case, keep it. It doesn't really hurt.